src/question/order_select/mod.rs

```diff
@@ -0,0 +1,305 @@
+mod builder;
+
+use std::io;
+
+use ui::{
+    backend::Backend,
+    events::{EventIterator, KeyEvent},
+    style::Color,
+    widgets::{self, Text},
+    Prompt, Widget,
+};
+
+use crate::{Answer, Answers, ListItem};
+
+use super::{
+    handler::{Filter, Transform, Validate}, choice::SelectList,
+};
+
+pub use builder::OrderSelectBuilder;
+
+#[cfg(test)]
+mod tests;
+
+// =============================================================================
+//
+// =============================================================================
+
+#[derive(Debug)]
+pub(super) struct OrderSelect<'a> {
+    choices: SelectList<OrderSelectItem>,
+    max_index_width: usize,
+    moving: bool,
+
+    transform: Transform<'a, [OrderSelectItem]>,
+    validate: Validate<'a, [OrderSelectItem]>,
+    filter: Filter<'a, Vec<OrderSelectItem>>,
+}
+
+impl<'a> Default for OrderSelect<'a> {
+    fn default() -> Self {
+        Self { 
+            choices: SelectList::new(|_| true),
+
+            // can't put
+            // `..Default::default()`
+            // because of recursion
+            max_index_width: Default::default(),
+            moving: Default::default(),
+            transform: Default::default(),
+            validate: Default::default(),
+            filter: Default::default(),
+        }
+    }
+} 
+
+impl widgets::List for OrderSelect<'_> {
+    fn render_item<B: ui::backend::Backend>(
+        &mut self,
+        index: usize,
+        hovered: bool,
+        mut layout: ui::layout::Layout,
+        b: &mut B,
+    ) -> std::io::Result<()> {
+        let symbol_set = ui::symbols::current();
+
+        if hovered {
+            if self.moving {
+                b.set_bg(Color::Cyan)?;
+                b.set_fg(Color::Black)?;
+            } else {
+                b.set_fg(Color::Cyan)?;
+            }
+
+            write!(b, "{} ", symbol_set.pointer)?;
+        } else {
+            b.write_all(b"  ")?;
+        }
+
+        write!(
+            b,
+            "{:>width$}. ",
+            index,
```

Comment:
I'm slightly confused about what `index` is supposed to represent here.

If the current position is meant to be rendered here then this is correct, but there is a disconnect between what is rendered, and the `index` in `OrderSelectItem`. If this is the intended behaviour, there should be some documentation stating the difference.

If the starting `index` is meant to be rendered, then this should be `self.choices[index].index`.

Reply:
I think leaving it as `index` is fine. Just make add a note in the documentation, that it refers to the index it was initially at.

Reply:
Actually, I just read the change you made, and `initial_index` is a good name.